internal/audit/kafka/kafka_test.go

```diff
@@ -152,12 +202,71 @@ func TestAsyncProduce(t *testing.T) {
 
 	// validate we see this entries in kafka, eventually
 	require.Eventually(t, func() bool {
-		records, err := fetchKafkaTopic(uri, defaultIntegrationTopic)
+		records, err := fetchKafkaTopic(uri, defaultIntegrationTopic, false)
 		require.NoError(t, err)
 		return len(records) == 2
 	}, 10*time.Second, 100*time.Millisecond, "expected to see audit log entries in kafka")
 }
 
+func newKafkaBrokerWithTLS(t *testing.T, topic, caPath, certPath, keyPath string) string {
+	t.Helper()
+
+	testDataAbsPath, err := filepath.Abs("testdata/valid")
+	require.NoError(t, err)
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err, "Failed to connect to Docker")
+
+	hostPort := 65136
+
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: redpandaImage,
+		Tag:        redpandaVersion,
+		Cmd: []string{
+			"redpanda",
+			"start",
+			"--mode", "dev-container",
+			// kafka admin client will retrieve the advertised address from the broker
+			// so we need it to use the same port that is exposed on the container
+			"--config", "/etc/redpanda/rpconfig.yaml",
+			"--advertise-kafka-addr", fmt.Sprintf("localhost:%d", hostPort),
+		},
+		ExposedPorts: []string{
+			"9092/tcp",
+		},
+		PortBindings: map[docker.Port][]docker.PortBinding{
+			"9092/tcp": {{HostIP: "localhost", HostPort: strconv.Itoa(hostPort)}},
+		},
+		Mounts: []string{
+			fmt.Sprintf("%s:/etc/redpanda", testDataAbsPath),
+		},
+	}, func(config *docker.HostConfig) {
+		config.AutoRemove = true
+	})
+	require.NoError(t, err, "Failed to start container")
+
+	t.Cleanup(func() {
+		_ = pool.Purge(resource)
+	})
+
+	brokerDSN := fmt.Sprintf("localhost:%d", hostPort)
+	duration := 10 * time.Second
+	skipVerify := false
+	tlsConfig, err := kafka.NewTLSConfig(context.Background(), duration, skipVerify, caPath, certPath, keyPath)
```

Comment:
I think this leaks a goroutine because the context is never canceled.

```suggestion
	ctx, cancel := context.WithCancel(context.Background())
	t.Cleanup(cancel)
	tlsConfig, err := kafka.NewTLSConfig(ctx, duration, skipVerify, caPath, certPath, keyPath)
```

(alternatively we could set the reload duration to 0, if we allow that as suggested)